HW0/P4/.ipynb_checkpoints/CS 205 HW0 P4-checkpoint.ipynb

```diff
@@ -0,0 +1,256 @@
+    "    g = -9.81  # Gravity\n",
+    "    # For constructing matrix B\n",
+    "    bx = 0\n",
+    "    by = 0\n",
+    "    bz = 0\n",
+    "    bvx = 0.25\n",
+    "    bvy = 0.25\n",
+    "    bvz = 0.1\n",
+    "    # For constructing matrix C\n",
+    "    rx = 1.0\n",
+    "    ry = 5.0\n",
+    "    rz = 5.0\n",
+    "    rvector=[rx, ry, rz]\n",
+    "\n",
+    "    # Create 3D axes for plotting\n",
+    "    ax = Axes3D(plt.figure())\n",
+    "    \n",
+    "#####################\n",
+    "    # Part 1:\n",
+    "    # Load true trajectory and plot it\n",
+    "data = np.loadtxt('P4_trajectory.txt',delimiter=',')\n",
+    "\n",
+    "print (len(data))\n",
+    "\n",
+    "xcoords=[]\n",
+    "ycoords=[]\n",
+    "zcoords=[]\n",
+    "vxcoords=[]\n",
+    "vycoords=[]\n",
+    "vzcoords=[]\n",
+    "\n",
+    "\n",
+    "for i in range(len(data)):\n",
+    "    xcoords.append(data[i][0])\n",
+    "    ycoords.append(data[i][1])\n",
+    "    zcoords.append(data[i][2])\n",
+    "    vxcoords.append(data[i][3])\n",
+    "    vycoords.append(data[i][4])\n",
+    "    vzcoords.append(data[i][5])\n",
+    "    \n",
+    "# Normally, this data wouldn't be available in the real world\n",
+    "    #####################\n",
+    "\n",
+    "ax.plot(xcoords, ycoords, zcoords,'-b', label='True trajectory')\n",
+    "\n",
+    "#####################\n",
+    "    # Part 2:\n",
+    "data2 = np.loadtxt('P4_measurements.txt',delimiter=',')\n",
+    "\n",
+    "# Read the observation array and plot it (Part 2)\n",
+    "xcoords2=[]\n",
+    "ycoords2=[]\n",
+    "zcoords2=[]\n",
+    "\n",
+    "\n",
+    "for i in range(len(data2)):\n",
+    "    xcoords2.append(data2[i][0]/rx)\n",
+    "    ycoords2.append(data2[i][1]/ry)\n",
+    "    zcoords2.append(data2[i][2]/rz)\n",
+    "    \n",
+    "ax.plot(xcoords2, ycoords2, zcoords2,'.g', label='Observed trajectory')  \n",
+    "\n",
+    "#####################\n",
+    "    # Part 3:\n",
+    "    # Use the initial conditions and propagation matrix for prediction\n",
+    "#####################\n",
+    "\n",
+    "a = np.zeros(shape=(1,6))\n",
+    "a[0][5]=g*dt\n",
+    "a=np.asmatrix(a)\n",
+    "\n",
+    "    # Initial conditions for s0\n",
+    "s0 = [0, 0, 2, 15, 3.5, 4]\n",
+    "s0=np.asmatrix(s)\n",
+    "s0=np.transpose(s0)\n",
+    "\n",
+    "\n",
+    "# Compute the rest of sk using Eq (1)\n",
+    "\n",
+    "A=np.zeros(shape=(6,6))\n",
+    "for i in range(3):\n",
+    "    A[i][i]=1\n",
+    "    \n",
+    "for i in range(3,6):\n",
+    "    A[i][i]=1-dt*c\n",
+    "    A[i-3][i]=dt\n",
+    "    \n",
+    "A=np.asmatrix(A)\n",
+    "\n",
+    "sk=np.zeros(shape=(len(data),6))\n",
+    "sk=np.asmatrix(sk)\n",
+    "\n",
+    "\n",
+    "sk[0,:]=np.transpose(s0)\n",
+    "xcoords3=[]\n",
+    "ycoords3=[]\n",
+    "zcoords3=[]\n",
+    "for i in range((len(data)-1)):\n",
+    "    sk[i+1,:]= np.transpose(A*(np.transpose(sk[i,:]))) + a\n",
+    "\n",
```

Comment:
I appreciate the compactness of using the sk matrix for old value, new value, and value store. On the other hand, it definitely took me a little effort to understand what you had going on. Might be a good place to add a comment to help your reader